### PR TITLE
Created Header for all screens

### PR DESCRIPTION
**Pull request by bryanansong**
## Description
I created a header component that has a search bar for looking up exercises. In future updates, I would want to make this go through the exercises from our API

<what this pull request is doing>
<what will be done in later PRs and not included here>

## Test Plan
Screenshot 2024-06-28 at 3 47 49 PM

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: frontend/src/components/WorkoutTemplateCard/WorkoutTemplateCard.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+const WorkoutTemplateCard: React.FC<WorkoutTemplateCardProps> = ({
+  template,
+}) => {
+  // TODO: Implement onclick method to open a workout session page
+  return (
+    <div className={styles.container} onClick={() => {}}>
+      <p className={styles.cardTitle}>
+        {template ? template.name : "No name yet"}
```
Comment (cherieliu) [inline frontend/src/components/WorkoutTemplateCard/WorkoutTemplateCard.tsx:20]: will the ``template`` prop ever be null?

**Review by cherieliu — COMMENTED** [review]
lgtm!! great to see you building out more specified components for the dashboard

File: frontend/src/components/WorkoutTemplateCard/WorkoutTemplateCard.tsx (hunk 1 above)
Comment (bryanansong) [reply frontend/src/components/WorkoutTemplateCard/WorkoutTemplateCard.tsx:20]: No template name will never be null. I did this because I didn't have any data to work with yet but I wanted to see some text to be able to style the template title. When I start working with data, I can replace this with just `{ template }` or if I somehow forget to replace it, the correct name will still be displayed always.

**Comment by m5matthew** [comment]
Curious, so when a user searches an exercise and clicks on one, what is the expected user flow?

**Comment by bryanansong** [comment]
> Curious, so when a user searches an exercise and clicks on one, what is the expected user flow?

I'm still thinking about that
